src/frontend/src/eth/derived/network.derived.ts

```diff
@@ -1,13 +1,13 @@
+import { SUPPORTED_ETHEREUM_NETWORKS } from '$env/networks.env';
 import { enabledEthereumNetworks } from '$eth/derived/networks.derived';
 import type { EthereumNetwork } from '$eth/types/network';
-import { DEFAULT_NETWORK } from '$lib/constants/networks.constants';
 import { networkId } from '$lib/derived/network.derived';
 import { derived, type Readable } from 'svelte/store';
 
 export const selectedEthereumNetwork: Readable<EthereumNetwork> = derived(
 	[enabledEthereumNetworks, networkId],
 	([$enabledEthereumNetworks, $networkId]) =>
-		$enabledEthereumNetworks.find(({ id }) => id === $networkId) ?? DEFAULT_NETWORK
+		$enabledEthereumNetworks.find(({ id }) => id === $networkId) ?? SUPPORTED_ETHEREUM_NETWORKS[0]
```

Comment:
The original behavior was that

```typescript
const [DEFAULT_NETWORK, ...rest] = SUPPORTED_ETHEREUM_NETWORKS
```

Meaning that, in this part of code, it would always get `ETHEREUM_NETWORK`, and when it is disabled through `ETH_MAINNET_ENABLED`, it would get `SEPOLIA_NETWORK`.

However, it seems incorrect to do it via the variable `DEFAULT_NETWORK`, instead of doing directly via `SUPPORTED_ETHEREUM_NETWORKS`, according to the code context.

To be honest, maybe it should have been done in another PR, since it is a separate refactoring of an unnecessary link.

As alternative we can put there:

```typescript
$enabledEthereumNetworks.find(({ id }) => id === $networkId) ?? ETH_MAINNET_ENABLED ? ETHEREUM_NETWORK : SEPOLIA_NETWORK
```

What do you think?

1) I do in another PR
2) I leave it in this PR
3) I leave it in its original state (not really sure it is correct in this case)

Reply:
I'm not sure what's the issue (early morning, probably still need a coffee) but, if there is one, then I would do it in another PR as it does not seem related to the title of this PR.

Reply:
Done: created PR #1433